lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -87,7 +87,7 @@ public function get_items_permissions_check( $request ) {
 	public function get_items( $request ) {
 
 		//Make sure a search string is set in case the orderby is set to 'relevace'
-		if ( ! empty( $request['orderby'] ) && 'relevance' === $request['orderby'] && empty( $request['search'] ) ) {
+		if ( ! empty( $request['orderby'] ) && 'relevance' === $request['orderby'] && empty( $request['search'] ) && empty( $request['filter']['s'] ) ) {
 			return new WP_Error( 'rest_no_search_term_defined', __( 'You need to define a search term in order to use the relevance search.' ), array( 'status' => 400 ) );
```

Comment:
Could we change the error message to `'You need to define a search term to order by relevance.'`? Not a huge deal, but that would be a bit more clear to me, unless you feel the current message is better, just leave it as is.